src/posts.rs

```diff
@@ -134,11 +129,36 @@ impl Post {
             contents,
             url,
             published,
+            updated,
             release,
             layout,
             has_team: team.is_some(),
             team: team.unwrap_or_default(),
             team_url: team_url.unwrap_or_default(),
         })
     }
+
+    pub fn set_updated(&mut self, hour: u32) {
+        self.updated = build_post_time(self.year, self.month, self.day, hour);
+    }
+}
+
+fn build_post_time(year: u32, month: u32, day: u32, seconds: u32) -> String {
+    // build the time. this is only approximate, which is fine.
+    let mut time = Tm {
+        tm_sec: 0,
+        tm_min: 0,
+        tm_hour: 0,
+        tm_mday: day as i32,
+        tm_mon: (month as i32) - 1,    // 0-11 not 1-12
+        tm_year: (year as i32) - 1900, // from the year 1900, not the actual year
+        // these next two fields are wrong but we never use them to generate our times
+        tm_wday: 1,
+        tm_yday: 1,
+        tm_isdst: 0,
+        tm_utcoff: 0,
+        tm_nsec: 0,
+    };
+    time = time + Duration::seconds(seconds as i64);
```

Comment:
hm, can we assert that these seconds aren't putting us more than a day off?

Reply:
I added a check